src/Plugin/Field/FieldFormatter/InstagramEmbedFormatter.php

```diff
@@ -34,10 +38,16 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
     foreach ($items as $delta => $item) {
       $matches = [];
       foreach (Instagram::$validationRegexp as $pattern => $key) {
-        if (preg_match($pattern, $item->value, $matches)) {
+        if (preg_match($pattern, $this->getEmbedCode($item), $item_matches)) {
+          $matches[] = $item_matches;
           break;
         }
       }
+
+      if (!empty($matches)) {
+        $matches = reset($matches);
```

Comment:
Hm... I apologoze if my previous comment was unclear. The fact that you breat after first match means that you don't need this reset(). Just set $matches in the loop above and you should be fine.